File: src/main/java/led/SwingRenderer.java
package led;

import com.github.iamniklas.liocore.led.LEDRenderer;

import javax.swing.*;
import java.awt.*;

public class SwingRenderer extends LEDRenderer {

    private JLabel[] leds;

    public SwingRenderer(JLabel[] uiLeds, int _stripSize) {
        super(_stripSize);
        leds = uiLeds;
    }

    @Override
    public void render() {
        for (int i = 0; i < leds.length; i++) {
            leds[i].setForeground(new Color(colorData[i].r, colorData[i].g, colorData[i].b));
        }
    }
}

File: src/main/java/led/SwingRenderer.kt
package led

import javax.swing.JLabel
import com.github.iamniklas.liocore.led.LEDRenderer
import com.github.iamniklas.liocore.led.colorspace.LIOColor
import java.awt.Color
import java.util.ArrayList

class SwingRenderer(private val leds: Array<JLabel?>) : LEDRenderer() {
    override fun render(_colorData: ArrayList<LIOColor>?) {
        for (i in leds.indices) {
            leds[i]!!.foreground = Color(_colorData!![i].r, _colorData[i].g, _colorData[i].b)
        }
    }
}